receipt/src/test/java/com/hyperwallet/android/receipt/repository/ReceiptDataSourceTest.java

```diff
@@ -0,0 +1,384 @@
+        assertThat(mReceiptDataSource.isFetchingData().getValue(), is(false));
+    }
+
+    @Test
+    public void testLoadInitial_returnNoReceipt() {
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) {
+                HyperwalletListener listener = (HyperwalletListener) invocation.getArguments()[1];
+                listener.onSuccess(null);
+                return listener;
+            }
+        }).when(mHyperwallet).listReceipts(any(ReceiptQueryParam.class),
+                ArgumentMatchers.<HyperwalletListener<HyperwalletPageList<Receipt>>>any());
+
+        // test
+        mReceiptDataSource.loadInitial(mInitialParams, mInitialCallback);
+
+        verify(mHyperwallet).listReceipts(any(ReceiptQueryParam.class),
+                ArgumentMatchers.<HyperwalletListener<HyperwalletPageList<Receipt>>>any());
+        verify(mInitialCallback, never()).onResult(ArgumentMatchers.<Receipt>anyList(), anyInt(), anyInt());
+
+        assertThat(mReceiptDataSource.getErrors().getValue(), is(nullValue()));
+        assertThat(mReceiptDataSource.isFetchingData().getValue(), is(false));
+    }
+
+    @Test
+    public void testLoadInitial_withError() {
+        final HyperwalletError error = new HyperwalletError("test message", "TEST_CODE");
+        List<HyperwalletError> errorList = new ArrayList<>();
+        errorList.add(error);
+        final HyperwalletErrors errors = new HyperwalletErrors(errorList);
+
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) {
+                HyperwalletListener listener = (HyperwalletListener) invocation.getArguments()[1];
+                listener.onFailure(new HyperwalletException(errors));
+                return listener;
+            }
+        }).when(mHyperwallet).listReceipts(any(ReceiptQueryParam.class),
+                ArgumentMatchers.<HyperwalletListener<HyperwalletPageList<Receipt>>>any());
+
+        // test
+        mReceiptDataSource.loadInitial(mInitialParams, mInitialCallback);
+
+        verify(mHyperwallet).listReceipts(any(ReceiptQueryParam.class),
+                ArgumentMatchers.<HyperwalletListener<HyperwalletPageList<Receipt>>>any());
+        verify(mInitialCallback, never()).onResult(ArgumentMatchers.<Receipt>anyList(), anyInt(), anyInt());
+
+        assertThat(mReceiptDataSource.getErrors().getValue(), is(notNullValue()));
+        assertThat(mReceiptDataSource.getErrors().getValue().getErrors(), Matchers.<HyperwalletError>hasSize(1));
+        assertThat(mReceiptDataSource.getErrors().getValue().getErrors().get(0).getCode(), is("TEST_CODE"));
+        assertThat(mReceiptDataSource.getErrors().getValue().getErrors().get(0).getMessage(), is("test message"));
+        assertThat(mReceiptDataSource.isFetchingData().getValue(), is(false));
+    }
+
+    @Test
+    public void testRetry_LoadInitial() {
```

Comment:
it should be testRetry_loadInitial - need to change the L upper case, same with testRetry_LoadAfter

Reply:
testRetry_LoadAfter is still with L upper case